src/Config/ObjectModel/PaginationOptions.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Service.Exceptions;
+
+namespace Azure.DataApiBuilder.Config.ObjectModel;
+
+/// <summary>
+/// Pagination options for the dab setup.
+/// Properties are nullable to support DAB CLI merge config
+/// expected behavior.
+/// </summary>
+public record PaginationOptions
+{
+    /// <summary>
+    /// Default page size.
+    /// </summary>
+    public const uint DEFAULT_PAGE_SIZE = 100;
+
+    /// <summary>
+    /// Max page size.
+    /// </summary>
+    public const uint MAX_PAGE_SIZE = 100000;
+
+    /// <summary>
+    /// The default page size for pagination.
+    /// </summary>
+    [JsonPropertyName("default-page-size")]
+    public uint? DefaultPageSize { get; init; } = null;
+
+    /// <summary>
+    /// The max page size for pagination.
+    /// </summary>
+    [JsonPropertyName("max-page-size")]
+    public uint? MaxPageSize { get; init; } = null;
+
+    [JsonConstructor]
+    public PaginationOptions(int? DefaultPageSize = null, int? MaxPageSize = null)
+    {
+        if (MaxPageSize is not null)
+        {
+            ValidatePageSize((int)MaxPageSize);
+            this.MaxPageSize = MaxPageSize == -1 ? Int32.MaxValue : (uint)MaxPageSize;
+            UserProvidedMaxPageSize = true;
+        }
+        else
+        {
+            this.MaxPageSize = MAX_PAGE_SIZE;
+        }
+
+        if (DefaultPageSize is not null)
+        {
+            ValidatePageSize((int)DefaultPageSize);
+            this.DefaultPageSize = DefaultPageSize == -1 ? (uint)this.MaxPageSize : (uint)DefaultPageSize;
+            UserProvidedDefaultPageSize = true;
+        }
+        else
+        {
+            this.DefaultPageSize = DEFAULT_PAGE_SIZE;
+        }
+
+        if (this.DefaultPageSize > this.MaxPageSize)
```

Comment:
I think this part of the validation could be added to the `ValidatePageSize` method, since a max page size can't be greater than itself, the comparison would only fail in the case of a default page size given that is larger than the max page size. Lets all the validation happen inside the validation method, and makes the name of that method more descriptive (as of now the validation in that method matches validation for max page size but not default)

Reply:
The idea was to have two seperate exceptions for size values being incorrect for the default page and for the max page size respectively. so the validate page size method, validates a single page size. Here we want to throw a different exception text, rather than just grouping it in one. I feel it is more clear this way.

Reply:
Two exceptions make sense for clarity, but you might want to rename this so that it reflects that it only validates the max size case.